### ISSUE TITLE
Pjax

### ISSUE THREAD

**Issue by jingoro** [suttacentral/legacy-suttacentral#66]
Pjax would be nice to have.

https://github.com/defunkt/jquery-pjax

**Comment by jingoro** [comment]
Basic tests have shown that this results in a about a 50% reduction in page loads. So a typical (post-cache) page load goes from about 1000ms -> 500ms.

Since page loads have a minimum floor, this would probably be much more effective if it was combined with some sort of caching scheme.

Probably best to wait on something this for now as the potential for complication is high.

**Comment by sujato** [comment]
@blake-sc @SamaneriVimala this is quite old now and the world moves on ... can we either confirm this as a suggestion or close it?

**Comment by ghost** [comment]
I think this can be closed. Ajax works fine and page loads are pretty quick. And we have not needed this in the last 3 years anyway.

**Comment by blake-sc** [comment]
I have done test implementations of this general principle. It is sometimes much faster especially with pre-caching. The main distinction is that there is no page load involved, this means the web browser doesn't need to tear down the entire page, then render a whole new page. Large parts of the page such as the header and footer don't need to be rendered again, and even though caching ensures most resources don't need to be downloaded to render a new page, it saves even the minimal effort of fetching the resource (such as font) from the cache.
Of course it also saves a bit of bandwidth, it always saves the cost of head, header and footer which is 5.6kb, sometimes this will be the bulk of the page size (as with an empty details page) in other cases insignificant (such as a full length text)

Except on really slow connections (in terms of bandwidth) it doesn't result in significantly faster page load in terms of internet connectivity, the main advantage is in terms of faster rendering which depends on the browser and device. Also pre-caching can be used which basically involves guessing which page the user will visit next and loading it in advance - note though that many web browsers will also pre-load links automatically if configured to (or if not configured not to).

**Comment by sujato** [comment]
Okay, so does this mean yay or nay?
